### ISSUE TITLE
Long Running Notebook Testing Support

### ISSUE THREAD

**Issue by harshad16** [opendatahub-io/kubeflow#151]
/kind feature

**Why you need this feature:**

With Notebook-controller changes, we need to be careful with making changes to the reconciliation loop, 
the change would result in restarting all the Notebooks already running in a cluster. 
As a user of the notebook, might be working/not saving work when the upgrade happens,
so this makes it a reason for concern. 

**Describe the solution you'd like:**
[A clear and concise description of what you want to happen.]

Include Tests for checking long-running notebooks to catch changes that would cause such issues.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

Previously, we have fallen into the trap as we were missing this kind of testing.
Example: 
Original inclusion: https://github.com/opendatahub-io/kubeflow/commit/41baf497ee3559890a6a5fa1f4ba4eb4a69c61a2
reverted due to the restart of notebooks: https://github.com/opendatahub-io/kubeflow/commit/588142611a3c6cbbe72a5ad7ca66f2aedab3dc22

**Comment by shalberd** [comment]
> the change would result in restarting all the Notebooks already running in a cluster

@atheo89 No, it would not. The image change trigger annotation is only added on newly-created notebooks, not to old notebooks. 

The controller only changes reconcile logic when annotation is set https://github.com/opendatahub-io/kubeflow/pull/133/files#diff-a436986c5e93996f9035eee6798d064ab8659957e8443eb44b3a4d3db289bde4R207

About the annotation: Done in odh-dashboard code ofr newly-created notebook CRs. Now, if someone were to deliberately modify an existing notebook via ODH dashboard, setting a different underlying imagestream name and tag, only then would the odh-dashboard annotation be added in a notebook CR. But I don't think that is your use case with long-running notebooks and the notebook controller.
